packages/backend/src/nest/auth/services/invites/invite.service.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Handles invite-related chain operations
+ */
+
+import { ChainServiceBase } from '../chainServiceBase'
+import { ValidationResult } from '@localfirst/crdx'
+import {
+  Base58,
+  FirstUseDevice,
+  InvitationState,
+  InviteResult,
+  Keyset,
+  ProofOfInvitation,
+  UnixTimestamp,
+} from '@localfirst/auth'
+import { SigChain } from '../../sigchain'
+import { RoleName } from '../roles/roles'
+import { createLogger } from '../../../common/logger'
+
+const logger = createLogger('auth:inviteService')
+
+export const DEFAULT_MAX_USES = 1
+export const DEFAULT_INVITATION_VALID_FOR_MS = 604_800_000 // 1 week
+
+class InviteService extends ChainServiceBase {
+  public static init(sigChain: SigChain): InviteService {
+    return new InviteService(sigChain)
+  }
+
+  public createUserInvite(
+    validForMs: number = DEFAULT_INVITATION_VALID_FOR_MS,
+    maxUses: number = DEFAULT_MAX_USES,
+    seed?: string
+  ): InviteResult {
+    const expiration = (Date.now() + validForMs) as UnixTimestamp
+    const invitation: InviteResult = this.sigChain.team.inviteMember({
+      seed,
+      expiration,
+      maxUses,
+    })
+    return invitation
+  }
+
+  public createDeviceInvite(validForMs: number = DEFAULT_INVITATION_VALID_FOR_MS, seed?: string): InviteResult {
+    const expiration = (Date.now() + validForMs) as UnixTimestamp
+    const invitation: InviteResult = this.sigChain.team.inviteDevice({
+      expiration,
+      seed,
+    })
+    return invitation
+  }
+
+  public revoke(id: string) {
+    this.sigChain.team.revokeInvitation(id)
+  }
+
+  public getById(id: Base58): InvitationState {
+    return this.sigChain.team.getInvitation(id)
+  }
+
+  public static generateProof(seed: string): ProofOfInvitation {
+    return SigChain.lfa.invitation.generateProof(seed)
+  }
+
+  public validateProof(proof: ProofOfInvitation): boolean {
```

Comment:
You can delete `validateProof`, `admitUser`, and `admitMemberFromInvite`.  This is handled by the LFA `Connection` class we use in libp2p.

Reply:
These are useful in testing so I'm going to leave them in for now I think